src/lib/handlers/drop.ts

```diff
@@ -3,21 +3,21 @@ import type { Handler } from './types';
 type Handle = ReturnType<Handler>;
 type Fn = Parameters<Handle>[0];
 
-const handler = (() => {
-	let running: null | Fn;
+const handler = (({ max = 1 }: { max?: number } = { max: 1 }) => {
+	const queue: Array<Fn> = [];
```

Comment:
I wonder if storing the actual perform is functional...maybe we could get around with a simple counter?

Reply:
Fair point. I do wonder if at some point we might want to have access to the instance of the task, but we can cross that bridge if we come to it